frame/dapps-staking/src/lib.rs

```diff
@@ -629,6 +617,37 @@ decl_module! {
                 }
             }
         }
+
+        #[weight = SimpleDispatchInfo::FixedNormal(1_000)]
+        fn claim(origin, era: EraIndex) {
+            let stash = ensure_signed(origin)?;
+            if let Some(active_era) = T::EraFinder::active() {
+                if era >= active_era.index {
+                    Err("cannot claim yet")?
+                }
+            }
+            let mut untreated_era = Self::untreated_eras(&stash);
+            if era == untreated_era {
+                Err("stash already rewarded")?
+            }
+            while era > untreated_era {
+                Self::propagate_total_stakes(&stash, &untreated_era, &(untreated_era + 1));
```

Comment:
It might be better to put a limit on the backward `Era`. There is a variable `history_depth` which does not allow to go back to an earlier era than its value.

Reply:
modified it in 5174549